### PR TITLE
Fix api response json structs

### LINKED ISSUE #19: Fix api response JSON structs
These have the API response JSON structs defined inside of a function:

https://github.com/skycoin/exchange-api/blob/master/exchange/c2cx.com/responses.go
https://github.com/skycoin/exchange-api/blob/master/exchange/cryptopia.co.nz/responses.go

They should be at the top level.  And the structs can implement a function that returns Balances()

`Balances()` should return `map[string]decimal.Decimal`.

The JSON fields that are float64 should be `decimal.Decimal` (https://godoc.org/github.com/shopspring/decimal)

In C2CX's case, "frozen" values must be subtracted from "balance" to get the spendable values. `Balances()` should return spendable values.  You must use the decimal library to parse and perform calculations with floating point values.

### PR DESCRIPTION
**Pull request by briancaine**
Fixes #19

### CODE REVIEW

**Comment by gz-c** [comment]
merge conflicts

File: api/cli/c2cx.go [hunk 1]
```diff
@@ -25,19 +26,29 @@ func sumbitTradeCMD() cli.Command {
 				return errInvalidInput
 			}
 			var (
-				symbol        string
-				price, amount float64
+				symbol                             string
+				price, amount                      decimal.Decimal
+				stopLoss, takeProfit, triggerPrice decimal.Decimal
 			)
+			if stopLoss, err := decimal.NewFromString(stoploss); err != nil {
```
Comment (gz-c) [inline api/cli/c2cx.go]: the `stoploss`, `stopLoss` etc variables look mishandled here?

Comment (briancaine) [reply api/cli/c2cx.go]: I thought the best way to get the most accurate `decimal.Decimal` values was to parse them as string flags, and then pass the strings to `decimal.NewFromString`.

`stoploss` is the raw string parameter and `stopLoss` is the parsed `decimal.Decimal`.

I just [updated the code to be more explicit with the variable names](https://github.com/skycoin/exchange-api/pull/31/commits/d05e698b7139e5eb2f69d5825852c5a6f92ec825).

(Or should I do it a different way?)

**Comment by gz-c** [comment]
`dependency not vendored: github.com/shopspring/decimal`

Need to do `dep ensure; dep prune`

**Comment by gz-c** [comment]
You have to commit the vendor/ folder to the repository

**Comment by briancaine** [comment]
Oh, crap, sorry. Fixed.

**Comment by gz-c** [comment]
build still failing

**Comment by briancaine** [comment]
I fixed a typo. However, I didn't really expect this PR to rely on the tests passing, because that's in the [other  issue](https://github.com/skycoin/exchange-api/issues/29).

Should we merge the two issues together?

**Comment by gz-c** [comment]
Don't merge the issues. Get the test fixed first

**Comment by gz-c** [comment]
Some of the test failures are related to the changes here. Fix the tests in master first so we can tell them apart

**Comment by briancaine** [comment]
Will do.

**Comment by gz-c** [comment]
merge conflicts

**Comment by briancaine** [comment]
Yep, working on it now.

**Pull request merged**